COMET/misc_plugins/PlotScripts/analysis_scripts/IV_PQC.py

```diff
@@ -1,82 +1,670 @@
+Surface_recombination_velocity_average = []
 
 class IV_PQC:
-
     def __init__(self, data, configs):
+        # Do the analysis of the gate diode files as the last one.
+        if not all(analysis_type[0:6] == 'IV_GCD' for analysis_type in data[list(data.keys())[0]]['header'][2]):
+            data = self.file_order(data)
 
         self.log = logging.getLogger(__name__)
-        self.data = convert_to_df(data, abs=False, keys="all")
+        self.data = convert_to_df(data, abs=False, keys='all')
         self.config = configs
         self.df = []
         self.basePlots = None
+        self.basePlots_2 = None
         self.name = "IV_PQC"
         self.PlotDict = {"Name": "IV"}
+        self.capincluded = False
+        # Add different columns to the data frame;
+        self.data["columns"].insert(3, "CapacityCopy")
+        self.data["columns"].insert(4, "derivative")
+        self.data["columns"].insert(5, "derivative2")
+        self.data["columns"].insert(6, "1C2")
+        self.data["columns"].insert(7, "derivative1C2")
+        self.data["columns"].insert(8, "x")
+        self.data["columns"].insert(9, "N")
+        self.data["columns"].insert(10, "firstderivative_gate")
+        self.capincluded = True
         self.measurements = self.data["columns"]
-        self.xaxis = "voltage"
-        self.donts = ["timestamp", "voltage"]
-        #hv.renderer('bokeh')
-
-        c2 = 1 / (self.data['ChuckLeakage-Default_HME_S-PQC_Flute_1-Diode_IV']['data']["current"] *
-             self.data['ChuckLeakage-Default_HME_S-PQC_Flute_1-Diode_IV']['data']["current"])
-
+        self.xaxis = self.measurements[0] #Select your x axis, normally voltage.
         # Convert the units to the desired ones
         for meas in self.measurements:
             unit = self.config[self.name].get(meas, {}).get("UnitConversion", None)
             if unit:
                 self.data = convert_to_EngUnits(self.data, meas, unit)
 
-
-
     def run(self):
         """Runs the script"""
+        # Generate lists to store your plots
+        self.PlotDict["BasePlots_MOS"] = []
+        self.PlotDict["BasePlots_diode"] = []
+        self.PlotDict["BasePlots_gate"] = []
+
+        for df in self.data["keys"]:
+
+            # Start the cv_mos analysis
+            if self.data[df]['header'][1][0:6] == "CV_MOS": #check analysis type
+                self.PlotDict["BasePlots_MOS"] = self.analysis_mos(df)
+
+            # Start the Diode Analysis
+            elif self.data[df]['header'][1][0:8] == 'CV_Diode':
+                self.PlotDict["BasePlots_MOS"] = self.analysis_diode(df)
+
+            # Start the Gate diode analysis
+            else:
+                self.PlotDict["BasePlots_MOS"] = self.analysis_gate(df)
+
+        self.PlotDict["All"] = self.plot(self.PlotDict["BasePlots_diode"], self.PlotDict["BasePlots_MOS"], self.PlotDict["BasePlots_gate"], cv_files, diode_files, gate_files)
+
+        return self.PlotDict
+
+    def file_order(self, data):
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Method could be a function](https://app.codacy.com/manual/dominic.bloech/UniDAQ/pullRequest?prid=5481807)